packages/video_player_videohole/tizen/src/video_player.cc

```diff
@@ -0,0 +1,569 @@
+    if (geometry_configs_iter != geometry_configs->end()) {
+      if (std::holds_alternative<int>(
+              geometry_configs->at(flutter::EncodableValue("y")))) {
+        y_ = std::get<int>(geometry_configs->at(flutter::EncodableValue("y")));
+      }
+    }
+    geometry_configs_iter =
+        geometry_configs->find(flutter::EncodableValue("width"));
+    if (geometry_configs_iter != geometry_configs->end()) {
+      if (std::holds_alternative<int>(
+              geometry_configs->at(flutter::EncodableValue("width")))) {
+        width_ = std::get<int>(
+            geometry_configs->at(flutter::EncodableValue("width")));
+      }
+    }
+    geometry_configs_iter =
+        geometry_configs->find(flutter::EncodableValue("height"));
+    if (geometry_configs_iter != geometry_configs->end()) {
+      if (std::holds_alternative<int>(
+              geometry_configs->at(flutter::EncodableValue("height")))) {
+        height_ = std::get<int>(
+            geometry_configs->at(flutter::EncodableValue("height")));
+      }
+    }
+  }
+  LOG_INFO("[VideoPlayer] geometry of window,x:%d,y:%d,width:%d,height:%d", x_,
+           y_, width_, height_);
+}
+
+VideoPlayer::VideoPlayer(FlutterDesktopPluginRegistrarRef registrar_ref,
+                         const CreateMessage &create_message)
+    : registrar_ref_(registrar_ref) {
+  ParseCreateMessage(create_message);
+}
+
+bool VideoPlayer::Open(const std::string &uri) {
+  int ret = player_create(&player_);
+  if (ret != PLAYER_ERROR_NONE) {
+    LOG_ERROR("[VideoPlayer] player_create failed: %s", get_error_message(ret));
+    return false;
+  }
+
+  if (drm_type_ != DRM_TYPE_NONE) {
+    drm_manager_ =
+        std::make_unique<DrmManager>(drm_type_, license_url_, player_);
+    if (!drm_manager_->InitializeDrmSession(uri_)) {
+      LOG_ERROR("[VideoPlayer] initial drm session failed");
+    }
+  }
+
+  ret = player_set_uri(player_, uri.c_str());
+  if (ret != PLAYER_ERROR_NONE) {
+    player_destroy(player_);
+    LOG_ERROR("[VideoPlayer] player_set_uri failed: %s",
+              get_error_message(ret));
+    return false;
+  }
+  return true;
+}
+
+bool VideoPlayer::SetDisplay(FlutterDesktopPluginRegistrarRef registrar_ref) {
+  FlutterDesktopViewRef view_ref =
+      FlutterDesktopPluginRegistrarGetView(registrar_ref);
+  if (view_ref == nullptr) {
+    LOG_ERROR("[VideoPlayer] could not get window view handle");
+    return false;
+  }
+
+  int ret = 0;
+  void *lib_handle = dlopen("libcapi-media-player.so.0", RTLD_LAZY);
+  if (lib_handle) {
+    FuncPlayerSetEcoreWlDisplay player_set_ecore_wl_display =
+        reinterpret_cast<FuncPlayerSetEcoreWlDisplay>(
+            dlsym(lib_handle, "player_set_ecore_wl_display"));
+    if (player_set_ecore_wl_display) {
+      ret = player_set_ecore_wl_display(
+          player_, PLAYER_DISPLAY_TYPE_OVERLAY,
+          FlutterDesktopViewGetNativeHandle(view_ref), x_, y_, width_, height_);
```

Comment:
You're already using `dlopen` to call an internal API (`player_set_ecore_wl_display`) here. Wouldn't it make sense to use `dlopen` to get the address of `ecore_wl2_window_geometry_get` and use it to get the window geometry of the view, without having to pass the values from the Dart side?